sdk/search/search-documents/src/serviceUtils.ts

```diff
@@ -365,6 +370,54 @@ export function generatedIndexToPublicIndex(generatedIndex: GeneratedSearchIndex
   };
 }
 
+export function generatedSearchResultToPublicSearchResult<T>(results: GeneratedSearchResult[]) {
+  const returnValues: SearchResult<T>[] = [];
+  results.forEach((result) => {
+    const { _score, _highlights, ...restProps } = result;
+    const obj = {
+      score: _score,
+      highlights: _highlights,
+      document: {}
```

Comment:
can you either inline `...restProps` here or define `doc` before `obj` so you can use it directly?

Reply:
Done